homeassistant/components/config/zwave.py

```diff
@@ -196,3 +197,55 @@ def get(self, request, node_id):
                                       'label': value.label,
                                       'length': len(value.data)}
         return self.json(usercodes)
+
+
+class ZWaveProtectionView(HomeAssistantView):
+    """View for the protection commandclass of a node."""
+
+    url = r"/api/zwave/protection/{node_id:\d+}"
+    name = "api:zwave:protection"
+
+    async def get(self, request, node_id):
+        """Retrieve the protection commandclass options of node."""
+        nodeid = int(node_id)
+        hass = request.app['hass']
+        network = hass.data.get(const.DATA_NETWORK)
+        node = await hass.async_add_executor_job(network.nodes.get, nodeid)
```

Comment:
Don't add 5 jobs in 1 method to the executor pool, it's inefficient. It's better to add 1 big job to the executor pool.

```python
def gather_protection():
    node = network.nodes.get(nodeid)
    if node is None:
        return self.json_message(…)
    # …
    return self.json(…)

return await hass.async_add_executor_job(gather_protection)
```